File: 30 days of code/Day 23 - BST level-order traversal/BST level-order traversal.java
/**
 *              BST LEVEL-ORDER TRAVERSAL | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 23
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.*;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        {
            Node binaryTree = null;
            int nodes = scan.nextInt();
            while (nodes-- > 0) {
                int data = scan.nextInt();
                binaryTree = insert(binaryTree, data);
            }
            levelOrder(binaryTree);
        }
        scan.close();
    }

    private static Node insert(Node binaryTree, int data) {
        if (binaryTree == null) {
            return new Node(data);
        }

        Node tempNode;
        if (data <= binaryTree.data) {
            tempNode = insert(binaryTree.left, data);
            binaryTree.left = tempNode;
        }
        else {
            tempNode = insert(binaryTree.right, data);
            binaryTree.right = tempNode;
        }
        return binaryTree;
    }

    private static void levelOrder(Node binaryTree){
        Deque<Node> queue = new ArrayDeque<>();
        if (binaryTree != null) {
            queue.add(binaryTree);
            while (!queue.isEmpty()) {
                Node node = queue.poll();
                System.out.print(node.data + " ");

                if (node.left != null) {
                    queue.add(node.left);
                }
                if (node.right != null) {
                    queue.add(node.right);
                }
            }
        }
    }
}

class Node {
    Node left;
    Node right;
    int data;

    Node(int data) {
        this.data = data;
        left = right = null;
    }
}

File: 30 days of code/Day 23 - BST level-order traversal/BST level-order traversal.cs
/**
 *              BST LEVEL-ORDER TRAVERSAL | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 23
*/

using System;
using System.Collections.Generic;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        var reader = new InputReader(false);
        int totalNodes = reader.NextInt;
        Node tree = null;

        while (totalNodes-- > 0)
        {
            int data = reader.NextInt;
            tree = Insert(tree, data);
        }

        reader.Close();

        LevelOrder(tree);
    }

    static Node Insert(Node binaryTree, int data)
    {
        if (binaryTree == null)
        {
            return new Node(data);
        }

        Node tempNode;
        if (data <= binaryTree.Data)
        {
            tempNode = Insert(binaryTree.Left, data);
            binaryTree.Left = tempNode;
        }
        else
        {
            tempNode = Insert(binaryTree.Right, data);
            binaryTree.Right = tempNode;
        }

        return binaryTree;
    }

    static void LevelOrder(Node binaryTree)
    {
        var queue = new Queue<Node>();

        if (binaryTree != null)
        {
            queue.Enqueue(binaryTree);

            while (!queue.Count.Equals(0))
            {
                var node = queue.Dequeue();
                Console.Write($"{node.Data} ");

                if (node.Left != null)
                {
                    queue.Enqueue(node.Left);
                }
                if (node.Right != null)
                {
                    queue.Enqueue(node.Right);
                }
            }
        }
    }
}

class Node
{
    public Node Left { get; set; }
    public Node Right { get; set; }
    public int Data { get; set; }

    public Node(int data)
    {
        Data = data;
        Left = Right = null;
    }
}

class InputReader
{
    private TextReader input;

    public InputReader(bool fromFile)
    {
        input = fromFile ? new StreamReader("input.txt") : Console.In;
    }

    public string NextLine => input.ReadLine();

    public int NextInt => int.Parse(NextLine);

    public void Close() => input.Close();
}
